tests/api/utils/test_scheduler.py

```diff
@@ -1217,6 +1217,66 @@ async def test_schedule_job_concurrency_limit(
         assert call_counter == run_amount
 
 
+@pytest.mark.asyncio
+async def test_schedule_job_next_run_time(
+    db: Session,
+    scheduler: Scheduler,
+    k8s_secrets_mock: tests.api.conftest.K8sSecretsMock,
+):
+    """
+    This test checks that the next run time is updated after a schedule was skipped due to concurrency limit.
+    It creates a schedule that runs every second for 3 seconds with concurrency limit of 1.
+    The run takes 2 seconds to complete so the function should be triggered twice in that time frame.
+    While the 1st run is still running, manually invoke the schedule (should fail due to concurrency limit)
+    and check that the next run time is updated.
+    """
```

Comment:
wouldn't it be easier and more of a realistic scenario that we will run a function every x seconds but the method inside of it will take x+1 this will cause the schedule to be skipped and actually demonstrate better what we are experiencing.

Reply:
what you are suggesting is a different flow than the one in the bug.
if we don't invoke the function then you don't know if it was invoked by the scheduler because it depends on the APScheduler implementation (the scheduler has an internal implementation of concurrency limiting using `max_instances`)
I added the next run time check to an existing test which should cover the flow you mentioned